modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/PicklistObjectFieldBusinessType.java

```diff
@@ -177,9 +170,23 @@ public void predefineObjectFieldSettings(
 			ObjectStateFlow objectStateFlow =
 				objectFieldSetting.getObjectStateFlow();
 
-			_objectStateFlowLocalService.addObjectStateFlow(
-				newObjectField.getUserId(), newObjectField.getObjectFieldId(),
-				objectStateFlow.getObjectStates());
+			if (oldObjectField != null) {
+				ObjectStateFlow serviceBuilderObjectStateFlow =
```

Comment:
@G-Sa rename it to `oldObjectStateFlow` to follow the same pattern from `ObjectField oldObjectField`